### PR TITLE
Fix swap max ux issues

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes RAI-496

### CODE REVIEW

File: src/screens/ExchangeModal.js [hunk 1]
```diff
@@ -266,41 +276,63 @@ const ExchangeModal = ({
...
+  //   updateInputBalance();
+  // }, [
+  //   accountAddress,
+  //   inputAmount,
+  //   inputBalance,
+  //   inputCurrency,
+  //   isMax,
+  //   selectedGasPrice,
+  //   updateInputAmount,
+  // ]);
```
Comment (christianbaroni) [inline src/screens/ExchangeModal.js:334]: did you mean to leave this in?

Comment (brunobar79) [reply src/screens/ExchangeModal.js]: Forgot to push!

File: src/raps/actions/unlock.js [hunk 2]
```diff
@@ -9,7 +10,7 @@ import transactionTypes from '../../helpers/transactionTypes';
 import store from '../../redux/store';
 import { dataAddNewTransaction } from '../../redux/data';
 import { rapsAddOrUpdate } from '../../redux/raps';
-import { contractUtils, gasUtils, logger } from '../../utils';
+import { contractUtils, gasUtils, logger, AllowancesCache } from '../../utils';
```
Comment (jinchung) [inline src/raps/actions/unlock.js:13]:
```suggestion
import { AllowancesCache, contractUtils, gasUtils, logger } from '../../utils';
```

File: src/screens/ExchangeModal.js [hunk 3]
```diff
@@ -266,41 +276,29 @@ const ExchangeModal = ({
...
-    };
-
-    // we should recalculate the input amount value every time
-    // the max available changes(due to selectedGasPrice changing)
-    const isDifferent = !isEqual(inputBalance, inputAmount);
-
-    if (isMax && isDifferent) {
-      updateInputAmount(inputBalance, inputBalance, true, true);
+    if (
+      inputCurrency.symbol === 'ETH' &&
```
Comment (jinchung) [inline src/screens/ExchangeModal.js:282]:
```suggestion
      inputCurrency.address === 'eth' &&
```

Comment (jinchung) [reply src/screens/ExchangeModal.js:282]: In case somebody decides to name their shitcoin with symbol ETH :D

File: src/screens/ExchangeModal.js [hunk 4]
```diff
@@ -266,41 +276,29 @@ const ExchangeModal = ({
...
     }
-
-    updateInputBalance();
   }, [
-    accountAddress,
-    inputAmount,
-    inputBalance,
-    inputCurrency,
-    isMax,
+    inputCurrency.symbol,
```
Comment (brunobar79) [inline src/screens/ExchangeModal.js:289]:
```suggestion
inputCurrency.address
```

**Pull request merged**